### DISCUSSION TITLE
SQL Server compatibility level

### DISCUSSION THREAD

**Discussion by leeannepedersen** [quartznet/quartznet#1228] [Q&A] [answered]
Is the recommended SQL Server compatibility level documented per version? We are currently running v3.0.7 in compatibility level 100 (SQL 2008) and I would like to upgrade it to 130 (SQL 2016) at a minimum, but preferably 150 (SQL 2019). I'm hopeful that a newer cardinality estimator might solve some performance issues.

**Comment by lahma** [comment, marked as answer]
[Quartz uses very simple SQL statement syntax](https://github.com/quartznet/quartznet/blob/main/src/Quartz/Impl/AdoJobStore/StdAdoConstants.cs) because it needs to work in multiple different databases, [for SQL server only specific use case](https://github.com/quartznet/quartznet/blob/main/src/Quartz/Impl/AdoJobStore/SqlServerDelegate.cs) there is using of `SELECT TOP n`. So you should safely be able to upgrade to latest compatibility version. 

You should also check [release notes history](https://github.com/quartznet/quartznet/releases) and whether there has been performance improvements since 3.0.7 that might help in your case (there are some).
